check_logs.py

```diff
@@ -13,7 +14,36 @@
 def _fetch(title, url, dest):
     print_yellow("fetching %s from: %s" % (title, url))
     # TODO: use something more robust like python wget library.
-    urllib.request.urlretrieve(url, dest)
+    retries = 0
+    max_retries = 7
+    retry_codes = [500, 504]
+    while retries < max_retries:
+        try:
+            time.sleep(2**retries)
+            urllib.request.urlretrieve(url, dest)
+            retries += 1
```

Comment:
retries += 1 needs to be before the urllib.request otherwise it'll never increment on failures.